commons/src/main/java/com/powsybl/commons/reporter/TreeReporter.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.reporter;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+public class TreeReporter extends AbstractReporter {
+
+    private final List<TreeReporter> childReporters = new ArrayList<>();
+    private final List<Report> reports = new ArrayList<>();
+
+    public TreeReporter() {
+        super();
```

Comment:
Useless?

Reply:
Removed, forcing the user to provide at least key and name